src/common/core.h

```diff
@@ -11,6 +11,10 @@
 #ifndef _FLUX_CORE_H
 #define _FLUX_CORE_H
 
+#ifdef __cplusplus
+extern "C" {
+#endif
+
```

Comment:
yeah, its not necessary in this particular case.  I added them b/c we already had them in `schedutil.h` (`schedutil.h` is just a wrapper that `#includes` all the `schedutil/*.h` files).  

If we don't want them here, then I'd say we remove the `extern C` from `schedutil.h` for consistency.

Reply:
Yeah, for clarity I say we just wrap function prototypes in `extern "C"` instead of wrapping other includes. (i.e. move `extern "C"` to after all `#include` directives). Thus our "includes-only" headers won't need a useless `extern "C"`.

Reply:
ok, lets go with no `extern C` in those "wrapper includes" files.  I'll adjust everywhere and audit to make sure there weren't mistakes in other files.